docs/guide.rst

```diff
@@ -909,6 +909,26 @@ test result.
 
 __ https://github.com/teemtee/tmt/tree/main/tests/multihost/complete/data
 
+Rerunning failed tests of previously executed runs
+------------------------------------------------------------------
+
+Executing failed tests again after fixing them is now possible
+with `tmt run --again test --failed-only`.
+
+This is only possible when you have the run directory available
+and `--id` argument provided (or use `--last`) as it needs the data from execute step
+to select only failed test cases. After new execute step, tmt will
+again merge the results from the previous run with the new ones
+to keep all the data for full report.
+
+.. code-block:: shell
+
+    $ tmt run --all
```

Comment:
The `--all` is not needed here right (afaik it is implicit)?

```suggestion
    $ tmt run
```

Reply:
It would take steps from plans, so if the plan has all the steps mentioned then yes, otherwise to be sure it's better to have --all IMO

Reply:
I beg to differ. It should be same outcome regardless what is in plan metadata. E.g. This plan with just execute defined, when executed as 'tmt run'  runs discover even though it wasn't set in the plan at all.
```
$ cat plan.fmf
execute:
    how: tmt
$ tmt run
...
    discover
        how: shell
        summary: 0 tests selected
        warning: No tests found, finishing plan.
```

Reply:
However, if one specified some step subcomand on the _command line_ then `--all` is necessary as without it just specified steps are executed.

Reply:
tldr: for this documented usecase --all isn't necessary.